Src/FluidCaching/Node.cs

```diff
@@ -55,7 +55,7 @@ private void RegisterWithLifespanManager()
         /// <summary>
         /// Removes the object from node, thereby removing it from all indexes and allows it to be garbage collected
         /// </summary>
-        public void Remove()
+        public void RemoveFromBag()
```

Comment:
Not sure if this name is more correct. We're actually removing it from the cache, not just from a bag. This method is not called when a node is moved between bags.